vault/resource_gcp_secret_impersonated_account.go

```diff
@@ -0,0 +1,235 @@
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return err
+	}
+
+	path := d.Id()
+
+	backend, err := gcpSecretImpersonatedAccountBackendFromPath(path)
+	if err != nil {
+		return fmt.Errorf("invalid path %q for GCP secrets backend impersonated account: %s", path, err)
+	}
+
+	impersonatedAccount, err := gcpSecretImpersonatedAccountNameFromPath(path)
+	if err != nil {
+		return fmt.Errorf("invalid path %q for GCP Secrets backend impersonated account: %s", path, err)
+	}
+
+	log.Printf("[DEBUG] Reading GCP Secrets backend impersonated account %q", path)
+	resp, err := client.Logical().Read(path)
+	if err != nil {
+		return fmt.Errorf("error reading GCP Secrets backend impersonated account %q: %s", path, err)
+	}
+
+	log.Printf("[DEBUG] Read GCP Secrets backend impersonated account %q", path)
+	if resp == nil {
+		log.Printf("[WARN] GCP Secrets backend impersonated account %q not found, removing from state", path)
+		d.SetId("")
+		return nil
+	}
+
+	if err := d.Set("backend", backend); err != nil {
+		return err
+	}
+	if err := d.Set("impersonated_account", impersonatedAccount); err != nil {
+		return err
+	}
+
+	for _, k := range []string{"token_scopes", "service_account_email", "service_account_project"} {
+		v, ok := resp.Data[k]
+		if ok {
+			if err := d.Set(k, v); err != nil {
+				return fmt.Errorf("error reading %s for GCP Secrets backend impersonated account %q: %q", k, path, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func gcpSecretImpersonatedAccountUpdate(d *schema.ResourceData, meta interface{}) error {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return err
+	}
+
+	path := d.Id()
+
+	data := map[string]interface{}{}
+	gcpSecretImpersonatedAccountUpdateFields(d, data)
+
+	log.Printf("[DEBUG] Updating GCP Secrets backend impersonated account %q", path)
+
+	_, err = client.Logical().Write(path, data)
+	if err != nil {
+		return fmt.Errorf("error updating GCP Secrets backend impersonated account %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Updated GCP Secrets backend impersonated account %q", path)
+
+	return gcpSecretImpersonatedAccountRead(d, meta)
+}
+
+func gcpSecretImpersonatedAccountDelete(d *schema.ResourceData, meta interface{}) error {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return err
+	}
+
+	path := d.Id()
+
+	log.Printf("[DEBUG] Deleting GCP secrets backend impersonated account %q", path)
+	_, err = client.Logical().Delete(path)
+	if err != nil {
+		return fmt.Errorf("error deleting GCP secrets backend impersonated account %q", path)
+	}
+	log.Printf("[DEBUG] Deleted GCP secrets backend impersonated account %q", path)
+
+	return nil
+}
+
+func gcpSecretImpersonatedAccountUpdateFields(d *schema.ResourceData, data map[string]interface{}) {
+	if v, ok := d.GetOk("service_account_email"); ok {
+		data["service_account_email"] = v.(string)
+	}
+
+	if v, ok := d.GetOk("token_scopes"); ok {
+		data["token_scopes"] = v.(*schema.Set).List()
+	}
```

Comment:
We should also read `service_account_project` from the terraform config; currently that field is never being written to Vault